spring-cloud-starter-common-security-config-web/src/test/java/org/springframework/cloud/common/security/TestOAuthSecurityConfiguration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.common.security;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.common.security.support.OnOAuth2SecurityEnabled;
+import org.springframework.cloud.common.security.support.SecurityStateBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+
+/**
+ * Setup Spring Security OAuth for the Rest Endpoint.
+ *
+ * @author Corneil du Plessis
+ */
+@Configuration
+@Conditional(OnOAuth2SecurityEnabled.class)
+@Import(TestOAuthSecurityConfiguration.SecurityStateBeanConfig.class)
+public class TestOAuthSecurityConfiguration extends OAuthSecurityConfiguration {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
```

Comment:
Any reason to include this?

Reply:
We need to mimic the configuration of Dataflow and Skipper.

Reply:
Sure, but that is just `super.configure(http);` - we should be able to remove this method. It would be helpful to add your reply comment `We need to mimic the configuration of Dataflow and Skipper.` to the top-level javadoc as well.

Reply:
Updated comment